File: src/main/java/com/dochiri/hexagonal/application/user/port/in/RegisterUserUseCase.java
package com.dochiri.hexagonal.application.user.port.in;

import com.dochiri.hexagonal.application.user.dto.RegisterUserCommand;
import com.dochiri.hexagonal.application.user.dto.RegisterUserResult;

public interface RegisterUserUseCase {

    RegisterUserResult register(RegisterUserCommand registerUserCommand);

}

File: src/main/kotlin/com/dochiri/hexagonal/application/user/port/in/RegisterUserUseCase.kt
package com.dochiri.hexagonal.application.user.port.`in`

import com.dochiri.hexagonal.application.user.dto.RegisterUserCommand
import com.dochiri.hexagonal.application.user.dto.RegisterUserResult

interface RegisterUserUseCase {
    fun register(registerUserCommand: RegisterUserCommand): RegisterUserResult
}
